Sources/XcodeGenKit/SourceGenerator.swift

```diff
@@ -247,7 +247,7 @@ class SourceGenerator {
             let isOutOfBasePath = !path.absolute().string.contains(project.basePath.absolute().string)
 
             // has no valid parent paths
-            let isRootPath = isOutOfBasePath || path.parent() == project.basePath
+            let isRootPath = (isBaseGroup && isOutOfBasePath) || path.parent() == project.basePath
```

Comment:
The `isBaseGroup` param is a newer addition than the rest of this line. I wonder if that's all we need